### DISCUSSION TITLE
MountainDuck S3 integration, can you automate changing access key/secret key?

### DISCUSSION THREAD

**Discussion by TurboCooler** [iterate-ch/cyberduck#15728] [Q&A]
I would like to deploy MountainDuck in my organization, but we have a policy to rotate the access key/ secret key every 90 days.  Is there any way I can automate this, so the user does not have to know anything about access key/secret key rotations?  Is the information stored in a location that is accessible by a script?

**Comment by dkocher** [comment]
You have two possible options that may be feasible
* Connect using the credentials from the AWS command line tools [^1] 
* Instead of using long lived credentials switch to temporary tokens obtained from AWS STS in conjunction with a setup using OpenID Connect [^2] which we discussed in this [blog post](https://blog.cyberduck.io/2023/10/18/connect-to-s3-with-web-identity-federation/).

[^1]: https://docs.mountainduck.io/protocols/s3/#connecting-using-credentials-from-aws-command-line-interface 
[^2]: https://docs.mountainduck.io/protocols/s3/#connecting-with-openid-connect-oidc-identity-provider
